### ISSUE TITLE
sweet-egison failing to build for stackage

### ISSUE THREAD

**Issue by cdornan** [egison/egison#248]
`sweet-egison` is failing to build for stackage as follows (we have reverted to to 4.0.3 in the meantime):

```
Log files have been written to: /var/stackage/work/unpack-dir/.stack-work/logs/

--  While building package sweet-egison-0.1.1.0 using:
      /var/stackage/.stack/setup-exe-cache/x86_64-linux/Cabal-simple_mPHDZzAJ_3.2.0.0_ghc-8.10.2 --builddir=.stack-work/dist/x86_64-linux/Cabal-3.2.0.0 build lib:sweet-egison test:test bench:comb2 bench:perm2 --ghc-options ""
    Process exited with code: ExitFailure 1
    Logs have been written to: /var/stackage/work/unpack-dir/.stack-work/logs/sweet-egison-0.1.1.0.log

    Configuring sweet-egison-0.1.1.0...
    Preprocessing library for sweet-egison-0.1.1.0..
    Building library for sweet-egison-0.1.1.0..
    [1 of 6] Compiling Control.Egison.Matcher
    [2 of 6] Compiling Control.Egison.Match
    [3 of 6] Compiling Control.Egison.QQ

    /var/stackage/work/unpack-dir/unpacked/sweet-egison-0.1.1.0-d07deb9ad72699087362fd32e5ffd318324f44867861bd0d6f1c4628eee8e02c/src/Control/Egison/QQ.hs:192:55: error:
        • Couldn't match type ‘Exp’ with ‘Maybe Exp’
          Expected type: [Maybe Exp]
            Actual type: [Exp]
        • In the first argument of ‘TupE’, namely ‘pps’
          In the second argument of ‘AppE’, namely ‘(TupE pps)’
          In the first argument of ‘AppE’, namely
            ‘(AppE (VarE cName) (TupE pps))’
        |
    192 |                  (AppE (AppE (AppE (VarE cName) (TupE pps)) (VarE mName)) (VarE tName))
        |                                                       ^^^

    /var/stackage/work/unpack-dir/unpacked/sweet-egison-0.1.1.0-d07deb9ad72699087362fd32e5ffd318324f44867861bd0d6f1c4628eee8e02c/src/Control/Egison/QQ.hs:207:55: error:
        • Couldn't match type ‘Exp’ with ‘Maybe Exp’
          Expected type: [Maybe Exp]
            Actual type: [Exp]
        • In the first argument of ‘TupE’, namely ‘pps’
          In the second argument of ‘AppE’, namely ‘(TupE pps)’
          In the first argument of ‘AppE’, namely
            ‘(AppE (VarE cName) (TupE pps))’
        |
    207 |                  (AppE (AppE (AppE (VarE cName) (TupE pps)) (VarE mName)) (VarE tName))
        |                                                       ^^^
Progress 1363/1368
curator: Received ExitFailure 1 when running
Raw command: /usr/bin/stack --terminal --system-ghc build --test --bench --test-suite-timeout=600 --no-rerun-tests --no-run-benchmarks --haddock --no-interleaved-output --jobs=2
Run from: unpack-dir
```

**Comment by greymd** [comment]
It seems homebrew package is also failed due to this issue [homebrew-egison > GitHub Actions](https://github.com/egison/homebrew-egison/runs/1256519225?check_suite_focus=true).
GHC version is 8.10.1.

**Comment by egisatoshi** [comment]
@cdornan and @greymd 

Thank you for letting us know about the issue!
We will fix the problem soon.

**Comment by egisatoshi** [comment]
@cdornan and @greymd

I removed the compiling error and released the newest version 4.1.1.
Please try!
